sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -0,0 +1,286 @@
+
+            // When we have 'foo() && false', 'foo()' is the redundant part
+            CheckForBooleanConstantOnLeftReportOnInvertedLocation(binaryExpression, FalseLiteral, context);
+            CheckForBooleanConstantOnRightReportOnInvertedLocation(binaryExpression, FalseLiteral, context);
+        }
+
+        // LogicalOr (C#) / OrElse (VB)
+        protected void CheckOrExpression(SyntaxNodeAnalysisContext context)
+        {
+            var binaryExpression = (TBinaryExpression)context.Node;
+            if (CheckForNullabilityAndBooleanConstantsReport(binaryExpression, context, reportOnTrue: false))
+            {
+                return;
+            }
+
+            // When we have 'foo() || false', 'false' is the redundant part
+            CheckForBooleanConstantOnLeftReportOnExtendedLocation(binaryExpression, FalseLiteral, context);
+            CheckForBooleanConstantOnRightReportOnExtendedLocation(binaryExpression, FalseLiteral, context);
+
+            // When we have 'foo() || true', 'foo()' is the redundant part
+            CheckForBooleanConstantOnLeftReportOnInvertedLocation(binaryExpression, TrueLiteral, context);
+            CheckForBooleanConstantOnRightReportOnInvertedLocation(binaryExpression, TrueLiteral, context);
+        }
+
+        protected void CheckEquals(SyntaxNodeAnalysisContext context)
+        {
+            var binaryExpression = (TBinaryExpression)context.Node;
+            if (CheckForNullabilityAndBooleanConstantsReport(binaryExpression, context, reportOnTrue: true))
+            {
+                return;
+            }
+
+            CheckForBooleanConstantOnLeftReportOnExtendedLocation(binaryExpression, TrueLiteral, context);
+            CheckForBooleanConstantOnLeftReportOnNormalLocation(binaryExpression, FalseLiteral, context);
+
+            CheckForBooleanConstantOnRightReportOnExtendedLocation(binaryExpression, TrueLiteral, context);
+            CheckForBooleanConstantOnRightReportOnNormalLocation(binaryExpression, FalseLiteral, context);
+        }
+
+        protected void CheckNotEquals(SyntaxNodeAnalysisContext context)
+        {
+            var binaryExpression = (TBinaryExpression)context.Node;
+            if (CheckForNullabilityAndBooleanConstantsReport(binaryExpression, context, reportOnTrue: false))
+            {
+                return;
+            }
+
+            CheckForBooleanConstantOnLeftReportOnExtendedLocation(binaryExpression, FalseLiteral, context);
+            CheckForBooleanConstantOnLeftReportOnNormalLocation(binaryExpression, TrueLiteral, context);
+
+            CheckForBooleanConstantOnRightReportOnExtendedLocation(binaryExpression, FalseLiteral, context);
+            CheckForBooleanConstantOnRightReportOnNormalLocation(binaryExpression, TrueLiteral, context);
+        }
+
+        protected void CheckTernaryExpressionBranches(SyntaxNodeAnalysisContext context, SyntaxTree ternaryTree, SyntaxNode thenBranch, SyntaxNode elseBranch)
```

Comment:
[cosmetic] It's nice to split lines around 130 characters to allow easier review on GitHub.